packages/web-app-unzip/src/composables/useUnzipAction.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  dirname,
+  FileAction,
+  FileActionOptions,
+  formatFileSize,
+  UppyService,
+  useClientService,
+  useLoadingService,
+  useMessages,
+  useResourcesStore,
+  useService,
+  useUserStore
+} from '@ownclouders/web-pkg'
+import { computed, unref } from 'vue'
+import { useGettext } from 'vue3-gettext'
+import { extractNameWithoutExtension, urlJoin } from '@ownclouders/web-client'
+import { UppyFile } from '@uppy/core'
+import * as uuid from 'uuid'
+import * as zip from '@zip.js/zip.js'
+import PQueue from 'p-queue'
+
+const SUPPORTED_MIME_TYPES = ['application/zip']
+const MAX_SIZE = 64 // in mb
+
+export const useUnzipAction = () => {
+  const { $gettext, current: currentLanguage } = useGettext()
+  const clientService = useClientService()
+  const loadingService = useLoadingService()
+  const userStore = useUserStore()
+  const resourcesStore = useResourcesStore()
+  const { showErrorMessage } = useMessages()
+  const uppyService = useService<UppyService>('$uppyService')
+
+  const createRootFolder = async ({ space, resources }: FileActionOptions) => {
+    const archiveName = extractNameWithoutExtension(resources[0])
+
+    // conflict handling
+    let folderName: string
+    let conflictResolved = false
+    let iterations = 1
+    while (!conflictResolved) {
+      folderName = `${archiveName} (${iterations.toString()})`
+      iterations++
+
+      if (resourcesStore.resources.some(({ name }) => name === folderName)) {
+        continue
+      }
+
+      conflictResolved = true
+    }
+
+    const path = urlJoin(resourcesStore.currentFolder.path, folderName)
+    const resource = await clientService.webdav.createFolder(unref(space), { path })
+    resourcesStore.upsertResource(resource)
+    return resource
+  }
+
+  const getFileBlob = async ({ space, resources }: FileActionOptions) => {
+    const { response } = await clientService.webdav.getFileContents(
+      space,
+      { path: resources[0].path },
+      { responseType: 'blob' }
+    )
+    return new File([response.data], resources[0].name)
+  }
+
+  const handler = async ({ space, resources }: FileActionOptions) => {
+    let zipReader: zip.ZipReader<zip.BlobReader>
+
+    try {
+      zip.configure({ chunkSize: 128, useWebWorkers: true })
+      const fileBlob = await getFileBlob({ space, resources })
+      const blobReader = new zip.BlobReader(fileBlob)
+      zipReader = new zip.ZipReader(blobReader)
+      const entries = await zipReader.getEntries()
+
+      // encrypted archives currently cannot be extracted
+      if (entries.some(({ encrypted }) => encrypted)) {
+        showErrorMessage({
+          title: $gettext('Encrypted archives cannot be extracted'),
```

Comment:
```suggestion
          title: $gettext('Password protected archives cannot be extracted'),
```

for some users that are not that technical